libnetwork/ipams/defaultipam/structures.go

```diff
@@ -36,36 +45,85 @@ func (k SubnetKey) Is6() bool {
 // PoolIDFromString creates a new PoolID and populates the SubnetKey object
 // reading it from the given string.
 func PoolIDFromString(str string) (pID PoolID, err error) {
+	if strings.HasPrefix(str, poolIDV2Prefix) {
+		return parsePoolIDV2(str)
+	}
```

Comment:
May as well test and trim in the same operation.
```suggestion
	if v, ok := strings.CutPrefix(str, poolIDV2Prefix); ok {
		return parsePoolIDV2(v)
	}
```

Reply:
Yeah I considered that but decided to pass the unaltered string instead to have it in error messages.